lib/react_client.dart

```diff
@@ -41,13 +41,95 @@ newJsMap(Map map) {
 typedef JsObject ReactComponentFactory(Map props, [dynamic children]);
 typedef Component ComponentFactory();
 
-class ReactComponentFactoryProxy implements Function {
-  final ReactComponentFactory _call;
+/**
+ * A Dart function that creates React JS component instances.
+ */
+abstract class ReactComponentFactoryProxy implements Function {
+  /**
+   * The type of component created by this factory.
+   */
+  get type;
+
+  /**
+   * Returns a new rendered component instance with the specified props and children.
+   */
+  JsObject call(Map props, [dynamic children]);
+
+  /**
+   * Used to implement a variadic version of [call], in which children may be specified as additional options.
+   */
+  dynamic noSuchMethod(Invocation invocation);
+}
+
+/**
+ * A Dart function that creates React JS component instances for Dart components.
+ */
+class ReactDartComponentFactoryProxy extends ReactComponentFactoryProxy {
+  final JsFunction reactClass;
   final JsFunction reactComponentFactory;
-  ReactComponentFactoryProxy(this.reactComponentFactory, this._call);
+
+  ReactDartComponentFactoryProxy(JsFunction reactClass) :
+    this.reactClass = reactClass,
+    this.reactComponentFactory = _React.callMethod('createFactory', [reactClass]);
+
+  JsFunction get type => reactClass;
 
   JsObject call(Map props, [dynamic children]) {
-    return this._call(props, children);
+    List reactParams = [
+      generateExtendedJsProps(props, children),
+      children
+    ];
+
+    return reactComponentFactory.apply(reactParams);
+  }
+
+  dynamic noSuchMethod(Invocation invocation) {
+    if (invocation.memberName == #call && invocation.isMethod) {
+      Map props = invocation.positionalArguments[0];
+      List children = invocation.positionalArguments.sublist(1);
+
+      List reactParams = [generateExtendedJsProps(props, children)];
+      reactParams.addAll(children);
+
+      return reactComponentFactory.apply(reactParams);
+    }
+
+    return super.noSuchMethod(invocation);
+  }
+
+  /**
+   * Returns a JsObject version of the specified props, preprocessed for consumption by React JS
+   * and prepared for consumption by the react-dart wrapper internals.
+   */
+  static JsObject generateExtendedJsProps(Map props, dynamic children) {
+    if (children == null) {
+      children = [];
+    } else if (children is! Iterable) {
+      children = [children];
+    }
```

Comment:
@trentgrover-wf I noticed after the fact that I had kept this logic of normalizing `children` to a `List` intact, due to the order in which I refactored things.

Are you still of the opinion that it should be removed in this changeset?

It's not hurting things in here, and it would be nice for consumers to be able to switch over without having the functionality of `this.props['children']` being broken.

I'm not against removing it eventually, but I think it'd be better to do it in a separate PR.